parcel_test.go

```diff
@@ -31,63 +32,133 @@ func getTestParcel() Parcel {
 // TestAddGetDelete проверяет добавление, получение и удаление посылки
 func TestAddGetDelete(t *testing.T) {
 	// prepare
-	db, err := // настройте подключение к БД
+	db, err := sql.Open("sqlite", "tracker.db")
+	if err != nil {
+		t.Error(err)
+	}
+	defer db.Close()
+
+	_, err = db.Exec("DELETE FROM parcel") // - очистка
+	require.NoError(t, err)                // - очистка
+
 	store := NewParcelStore(db)
 	parcel := getTestParcel()
 
 	// add
-	// добавьте новую посылку в БД, убедитесь в отсутствии ошибки и наличии идентификатора
+	// добавьте новую посылку в БД, убедитесь в отсутствии ошибки и наличии идентификатора +
+
+	num, err := store.Add(parcel)
+	require.NoError(t, err)
+	assert.NotEmpty(t, num)
 
 	// get
 	// получите только что добавленную посылку, убедитесь в отсутствии ошибки
 	// проверьте, что значения всех полей в полученном объекте совпадают со значениями полей в переменной parcel
+	parcel.Number = num
+	p, err := store.Get(num)
+	require.NoError(t, err)
+	assert.Equal(t, parcel, p)
 
 	// delete
 	// удалите добавленную посылку, убедитесь в отсутствии ошибки
 	// проверьте, что посылку больше нельзя получить из БД
+	err = store.Delete(num)
+	require.NoError(t, err)
+
+	_, err = store.Get(num)
+	assert.Equal(t, sql.ErrNoRows, err)
```

Comment:
Функция `Equal` может сравнивать структуры сразу (а не по полям), надо только с полем `Number` разобраться.

Нужно переделать, так как при таком подходе при изменении структуры нужно будет менять все тесты

Reply:
не совсем понял в чем тут ошибка, здесь проверяю на отсутствие посылки в базе.

Reply:
Исправил в двух функциях: TestSetAddress, TestSetStatus

Reply:
Привет!
Да, ошибся комментарием

Reply:
Для сравнивания ошибок нужно использовать специализированную функцию
```suggestion
	require.ErrorIs(t, err, sql.ErrNoRows)
```